src/qibolab/_core/instruments/qm/program/instructions.py

```diff
@@ -136,6 +137,115 @@ def _process_sweeper(sweeper: Sweeper, args: ExecutionArguments):
     return variable, values
 
 
+def _qua_for_loop(variables: list[Variable], values: list[npt.NDArray]):
+    """Generate QUA ``for_`` loop command for the sweeps.
+
+    Partly copied from ``qualang_tools.from_array``.
+    """
+    if len(variables) > 1:
+        return for_each_(variables, values)
+
+    var = variables[0]
+    array = values[0]
+
+    if len(array) == 0:
+        raise ValueError("Sweeper values must have length > 0.")
+    elif len(array) == 1:
```

Comment:
The former branch is raising, no need for an `else`
```suggestion
    if len(array) == 1:
```